src/components/common/Form/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { Dispatch, SetStateAction, useState } from 'react';
```

Comment:
Consider usage of useForm hook in conjunction with yup() or any other joi.
JOI - in simple vords - validation scheme